platform/node/src/node_request.cpp

```diff
@@ -104,43 +118,34 @@ void NodeRequest::HandleCallback(const Nan::FunctionCallbackInfo<v8::Value>& inf
                     node::Buffer::Length(data)
                 );
             } else {
+                request->unref();
                 return Nan::ThrowTypeError("Response data must be a Buffer");
             }
         }
     }
 
-    if (request->ErrorMessage()) {
-        response.error = std::make_unique<mbgl::Response::Error>(
-            mbgl::Response::Error::Reason::Other,
-            request->ErrorMessage()
-        );
-    }
-
     // Send the response object to the NodeFileSource object
     callback(response);
+    request->unref();
     info.GetReturnValue().SetUndefined();
 }
 
-void NodeRequest::Execute() {
-    v8::Local<v8::Value> argv[] = { handle() };
-
-    Nan::AsyncResource res("mbgl:execute");
-    res.runInAsyncScope(Nan::To<v8::Object>(target->handle()->GetInternalField(1)).ToLocalChecked(), "request", 1, argv);
+void NodeRequest::unref() {
+  Nan::HandleScope scope;
```

Comment:
Why is this scope needed?

Reply:
I believe it is needed because asyncResource is a resource managed by node and we need to make sure that its removed as well from the memory managed by node.